### PR TITLE
[DOCU-2090] Encrypted fields in plugins

### PR DESCRIPTION
**Pull request by lena-larionova**
### Summary
Adding `encrypted: true` to all plugin parameters that will be encrypted if keyring db encryption is enabled.
* Entry in plugin layout to parse the new field
* Entry for new field in plugin template
* Bumping plugin versions in any place where the plugin had a minor or major version bump (eg 1.**4**.0 to 1.**5**.0, or **1**.0.0 to **2**.0.0)
  * Not bumping the version if it's only a patch level difference (1.5.**1**)
  * If the version in the doc is wildly different from the plugin's version, setting it to 2.7.x 
  * Fixing/adding some missing versions to the plugin data files (`version.yaml`)
* Adding changelog entries to plugin files for each field that is now encrypted

To do: 
* Add something to the upgrade guide about this being a breaking change.
* Update OIDC plugin

### Reason
Eng PRs: https://github.com/Kong/kong-ee/pull/2756 and https://github.com/Kong/kong/pull/8112

### Testing
To review, look at each plugin's `index.md` file; the numbered version files are the older content.

Sample edited plugins:
https://deploy-preview-3480--kongdocs.netlify.app/hub/kong-inc/session/
https://deploy-preview-3480--kongdocs.netlify.app/hub/kong-inc/basic-auth/
https://deploy-preview-3480--kongdocs.netlify.app/hub/kong-inc/vault-auth/ (only a changelog)

### CODE REVIEW

File: app/_hub/kong-inc/azure-functions/index.md [hunk 1]
```diff
@@ -202,3 +206,12 @@ HTTP/1.1 200 OK
...
 ```
+
+---
+
+## Changelog
+
+### 1.0.1
+
+* Starting with {{site.base_gateway}} 2.7.0.0, if keyring encryption is enabled,
+ the `apikey` and `clientid` parameter values will be encrypted.
```
Comment (kikito) [inline app/_hub/kong-inc/azure-functions/index.md:217]:
```suggestion
 the `config.apikey` and `config.clientid` parameter values will be encrypted.
```

File: app/_hub/kong-inc/vault-auth/index.md [hunk 2]
```diff
@@ -271,5 +263,14 @@ $ vault write kong-auth/foo - <<EOF
...
 ```
 
+
+## Changelog
+
+### 2.7.x
+> handler.lua version: 0.3.0
+
+* Starting with {{site.base_gateway}} 2.7.0.0, if keyring encryption is enabled
+and you are using Vault, the `vault_token` and `secret_token` will be encrypted.
```
Comment (kikito) [inline app/_hub/kong-inc/vault-auth/index.md:273]:
```suggestion
and you are using Vault, the `vaults.vault_token` and `vault_credentials.secret_token` fields will be encrypted.
```

**Review by kikito — CHANGES_REQUESTED** [review]
I have requested several minor changes, mostly renaming fields. I tried to add suggestions when possible.

A more important modification is that in three plugins (jwt-signer, kafka-log, kafka-upstream) the encryption will not work due to a bug. I have added notes about those.

Finally, I'd like to point out that we give a (now incomplete) list of encrypted fields in `gateway-2.7.x/app/gateway/2.7.x/plan-and-deploy/security/db-encryption.md`:

https://github.com/Kong/docs.konghq.com/blob/release/gateway-2.7.x/app/gateway/2.7.x/plan-and-deploy/security/db-encryption.md#encrypted-fields

I would suggest nuking that section completely, but that would involve documenting the fields in that section the same way they have been done here (by marking them as `encrypted=true` and adding a changelog entry).

File: app/_hub/kong-inc/oauth2/index.md [hunk 3]
```diff
@@ -498,3 +480,13 @@ Note that the rest of the credentials flow uses HTTPS and not gRPC protocol.  De
...
 [proxy-port]: https://docs.konghq.com/latest/configuration/#proxy_listen
+
+---
+
+## Changelog
+
+### 2.1.1
+
+* Starting with {{site.base_gateway}} 2.7.0.0, if keyring encryption is enabled
+and you are using OAuth2, the `provision_key` parameter value and the
```
[the remark this answers is not in the document]
Comment (lena-larionova) [reply app/_hub/kong-inc/oauth2/index.md:491]: Was this param name a typo, or is there another field that's encrypted? I based it on the original on the PR here, looks like `oauth2_credentials.client_secret` also gets encrypted: https://github.com/Kong/kong-ee/pull/2756/files#diff-24cfa06ad6f660eac509b208bfa2fb96fdf20a591a69b0b8adbd3bd44cbc82b2L42

File: app/_hub/kong-inc/openid-connect/index.md [hunk 4]
```diff
@@ -3305,3 +3308,22 @@ mean other gateways, load balancers, NATs, and such in front of Kong. If there i
...
+ parameter values will be encrypted.
+
+  Additionally, the `d`, `p`, `q`, `dp`, `dq`, `qi`, `oth`, `r`, `t`, and `k`
+  fields inside `openid_connect_jwks.previous[...].` and `openid_connect_jwks.keys[...]`
+  will be marked as encrypted.
+
+  {:.important}
+  > There's a bug in {{site.base_gateway}} that prevents keyring encryption
+  from working on deeply nested fields, so the `encrypted=true` setting does not
+  currently have any effect on the nested fields in this plugin.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/index.md:3329]: @kikito Is this accurate for the OIDC plugin? Looks like it should have the same issues as JWT signer.

File: app/gateway/2.7.x/plan-and-deploy/security/db-encryption.md [hunk 5]
```diff
@@ -280,11 +280,10 @@ When operating in `cluster` mode, keyring material propagates automatically amon
...
 
-* `key` fields of `certificate` objects (corresponding to the private key of a TLS certificate)
-* `password` fields of `basic-auth` plugin credential objects (note that passwords are also hashed by a one-way hashing function)
-* `key` fields of `key-auth-enc` plugin credential objects
-* `client_id` and `client_secret` fields of the `openid-connect` plugin
-* `aws_key` and `aws_secret` fields of the `aws-lambda` plugin
+* `key` fields of `certificate` objects (corresponding to the private key of a TLS certificate).
+* Certain configuration parameters in plugins and plugin-related authentication
+objects. For information on which plugin fields are encrypted, see
+[each individual plugin's documentation](/hub/).
```
Comment (lena-larionova) [inline app/gateway/2.7.x/plan-and-deploy/security/db-encryption.md:287]: Almost all of these fields (other than key in certificates) are already covered by this PR. I only had to add one more update for `key-auth-enc`. Let's split out further labelling changes, if any need to happen, into a separate PR.

File: app/_hub/kong-inc/oauth2/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kikito) [reply app/_hub/kong-inc/oauth2/index.md]: You are right, it was `oauth2_credentials.client_secret`, apologies

File: app/_hub/kong-inc/openid-connect/index.md (hunk 4 above)
Comment (kikito) [reply app/_hub/kong-inc/openid-connect/index.md:3329]: Correct, it will have the same issues. Well spotted.

**Review by kikito — APPROVED** [review]

**Pull request merged**